### DISCUSSION TITLE
React Native (iOS) : Error Build ipa for distribution

### DISCUSSION THREAD

**Discussion by dmpriatna** [codemagic-ci-cd/codemagic-docs#1785] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [x] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

i am following [this](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/react-native/react-native-demo-project/codemagic.yaml)

signing part was no trouble, and clearly that said key (profile) and certificate is imported
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/83553094/58f0e231-71b6-413f-951f-b3cb0f6cf7e2)

then setting Xcode part, i noticed that something not right.
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/83553094/815a2022-860c-4fa5-81ac-371a4ca86726)
output said there is no Profile, no Certificate, Method is ad-hoc (i am sure that choose distribution_type: app_store at my.yaml)

### Expected results

i thought result will be smooth until the *.ipa file uploaded to app store, but i got the error

**/Users/builder/clone/ios/myapp.xcodeproj: error: "myapp" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'myapp' from project 'myapp')**

can someone please help and guide me

### Actual results

this what blocked me
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/83553094/39408ac9-aeaa-482b-bb65-8c69885f62e6)

### Build id (optional)

648e8e88c23819a0da49dfc3

**Comment by himesh-cm** [comment, marked as answer]
Hi @dmpriatna ,
It seems `.xcodeproj` for the `RNImagePicker` is code signed instead of the actual project.

Please use the below change to your xcode-project use-profile command.

`xcode-project use-profiles --project ios/*.xcodeproj`

Re-run the build, and let us know how it goes.

**Reply by himesh-cm** [reply]
The thread was continued on intercom.
Along with the above changes, there was an issue with a typo in the bundle identifier.
